### PR TITLE
feat: Duffys

### PR DESCRIPTION
**Pull request by Romeo-PHILLIPS**
https://github.com/alltheplaces/alltheplaces/issues/3518

Opening hours a bit ugly but it works

### CODE REVIEW

File: locations/spiders/duffys.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    def start_requests(self):
+        url = "https://api.duffysmvp.com/api/app/nearByLocations"
+        yield scrapy.Request(
+            url,
+            method="POST",
+            headers={"Content-Type": "application/json", "Accept": "application/json"},
+            body='{"latitude":"26.6289791","longitude":"-80.0724384"}',
+            callback=self.parse,
+        )
```
Comment (Cj-Malone) [inline locations/spiders/duffys.py:22]:
```suggestion
        yield JsonRequest(
            url="https://api.duffysmvp.com/api/app/nearByLocations",
            data={"latitude": "26.6289791", "longitude": "-80.0724384"},
        )
```

**Comment by Cj-Malone** [comment]
Looks like the opening hours really working, AM/PM aren't taken into account so we get `opening_hours=Mo-Su 11:00-11:00` instead of `opening_hours=Mo-Su 11:00-23:00`. Also `ref=039` has `opening_hours=Su-We 1130:00-12:00; Th-Sa 1130:00-2:00`

**Comment by Romeo-PHILLIPS** [comment]
@Cj-Malone added super scary function that seems to fix it for all

**Comment by Cj-Malone** [comment]
Hope you don't mind, but it did indeed scare me.

**Review by Cj-Malone — APPROVED** [review]

**Pull request merged**